probe-rs/src/architecture/arm/core/m4.rs

```diff
@@ -239,8 +239,22 @@ impl From<FpRev1CompX> for u32 {
 }
 
 impl FpRev1CompX {
+    /// Get the correct comparator value stored at the given address
+    /// This will adjust the `FpRev1CompX.comp() result based on the `FpRev1CompX.replace()` specification
+    /// NOTE: Does not support a `replace value of '11'
+    fn get_breakpoint_comparator(register_value: u32) -> Result<u32, Error> {
+        let fp1_val = FpRev1CompX::from(register_value);
+        if fp1_val.replace() == 0b01 {
+            Ok(fp1_val.comp() << 2)
+        } else if fp1_val.replace() == 0b10 {
+            Ok((fp1_val.comp() << 2) & 0x2)
```

Comment:
I think this should be an or.

Reply:
@Tiwalun ... can you give me a bit more explanation please. Here's what I am trying to do. Shift the comparator to the left by two bits, then turn on the 0b10 bit, to create the address as `Breakpoint on instruction at '000':COMP:'10'.`

What am I missing?

Reply:
Absolutely correct ... aargh ... thanks :)